src/components/SlateEditor/plugins/embed/EditImage.jsx

```diff
@@ -1,73 +1,193 @@
+import Button from '@ndla/button';
+import { FormInput, StyledButtonWrapper } from '@ndla/forms';
+import { isEmpty } from '../../../validators';
 import ImageEditor from '../../../../containers/ImageEditor/ImageEditor';
 import { Portal } from '../../../Portal';
-import { EmbedShape } from '../../../../shapes';
+import { EmbedShape, EditorShape } from '../../../../shapes';
+import { getSchemaEmbed } from '../../editorSchema';
+
+const StyledInputWrapper = styled.div`
+  background: ${colors.brand.greyLightest};
+  padding: ${spacing.normal};
+  position: relative;
+  z-index: 20;
+`;
+
+const Background = styled.div`
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  background: rgba(0, 0, 0, 0.3);
+  z-index: 10;
+  ${animations.fadeIn()}
+`;
+
+const editorContentCSS = css`
+  box-shadow: ${shadows.levitate1};
+`;
 
 const imageEditorWrapperStyle = css`
   background-color: white;
 `;
 
 class EditImage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      alt: props.embed.alt,
+      caption: props.embed.caption,
+      imageUpdates: undefined,
+      madeChanges: false,
+    };
+    this.onUpdatedImageSettings = this.onUpdatedImageSettings.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onAbort = this.onAbort.bind(this);
+  }
+
   componentDidMount() {
     const { placeholderEl, embedEl } = this;
 
     const bodyRect = document.body.getBoundingClientRect();
 
-    const placeholderRect = placeholderEl.getBoundingClientRect();
-
+    // Use contenteditable as reference to fetch embed size when previewing.
+    const placeholderRect = placeholderEl
+      .closest('div[contenteditable="false"]')
+      .getBoundingClientRect();
     // Placing embed within placeholder div on mount
     embedEl.style.position = 'absolute';
     embedEl.style.top = `${placeholderRect.top - bodyRect.top}px`;
-    embedEl.style.left = `${placeholderRect.left}px`;
-    embedEl.style.width = `${placeholderRect.width}px`;
+    embedEl.style.left = `${placeholderRect.left +
+      spacing.spacingUnit -
+      placeholderRect.width * (0.333 / 2)}px`;
+    embedEl.style.width = `${placeholderRect.width * 1.333 -
+      spacing.spacingUnit * 2}px`;
 
     const embedRect = embedEl.getBoundingClientRect();
 
-    placeholderEl.style.height = `${embedRect.height}px`;
+    placeholderEl.style.height = `${embedRect.height +
+      spacing.spacingUnit * 2}px`;
+  }
+
+  onUpdatedImageSettings(imageUpdates) {
+    this.setState({
+      imageUpdates: imageUpdates,
+      madeChanges: true,
+    });
+  }
+
+  onSave() {
+    const { node, editor, onFigureInputMultipleUpdates } = this.props;
+
+    onFigureInputMultipleUpdates({
+      caption: this.state.caption,
+      alt: this.state.alt,
+    });
+
+    if (this.state.imageUpdates) {
+      const data = {
+        ...getSchemaEmbed(node),
+        ...this.state.imageUpdates.transformData,
```

Comment:
jvorfor ikke hente imageUpdates slik const { imageUpdates } = this.state. Så slipper du skrive this.state mange ganger